docs/source/configuration/authorization.mdx

```diff
@@ -98,199 +140,263 @@ So the `me` top level operation would not even be executed. When returning the r
-    orders: [User] @requiresScopes(["employee"])
+  id: ID
+  name: String
+  amount: Int @requiresScopes(["inventory:read"])
+  orders: [User] @requiresScopes(["user:read", "inventory:read"])
 }
 ```
 
-And this query:
+The router executes the following query differently, depending on the request's attached scopes:
 
 ```graphql
-{
-    me {
-        name
-        email
-    }
+query {
+  me {
+    name
+    email
+  }
 
-    products {
-        name
-        amount
-        orders {
-            name
-        }
+  products {
+    name
+    amount
+    orders {
+      name
     }
+  }
 }
 ```
 
-If the request presented the scopes `profile:read user:read`, then it would be interpreted as:
+If the request includes the `profile:read user:read` scope set, then the router would execute the following filtered query: 
 
+```graphql title="Request scopes: 'profile:read user:read'"
+query {
+  me {
+    name
+    email
+  }
 
-```graphql
-{
-    me {
-        name
-        email
-    }
+  products {
+    name
+    /* The following fields are filtered out:
+    amount # Requires "inventory:read" scope
+    orders # Requires "user:read" and "inventory:read" scopes
+    */
+  }
+}
+```
+
+The response would include errors at the `/products/@/amount` and `/products/@/amount` paths.
+
+If the request includes the `user:read inventory:read` scope set, then the router would execute the following filtered query: 
 
-    products {
-        name
+```graphql title="Request scopes: 'user:read inventory:read'"
+query {
+  /* The following fields are filtered out
+  me { # Requires the "profile:read" scope
+    name
+    email
+  }
+  */
+  products {
+    name
+    amount
+    orders {
+      name
     }
+  }
 }
 ```
 
-And generate an error at path `/products/@/amount`.
+The response would include an error at the `/me` path.
 
-While if it presented the scopes `employee inventory`, it would filter the query as:
+## Composition and federation
 
-```graphql
-{
-    products {
-        name
-        amount
-        orders {
-            name
-        }
-    }
+Authorization directives are defined at the subgraph level and the router composes them in the supergraph schema. In other words, if subgraph fields or types include either `@authenticated` or `@requiresScopes` directives, they are set on the supergraph too.
+
+If multiple shared subgraph fields include `@requiresScopes`, the supergraph schema merges the sets of scopes required by each subgraph. For example, imagine one subgraph requires the `profile:read` scope on the `users` query:
+
+```graphql title="Subgraph A"
+type Query {
+  users: [User!]! @requiresScopes(scopes: ["profile:read"])
 }
 ```
 
-And generate an error at path `/me`.
+And another subgraph requires the `user:read` scope on `users` query:
 
+```graphql title="Subgraph B"
+type Query {
+  users: [User!]! @requiresScopes(scopes: ["user:read"])
+}
+```
 
-## Composition and federation
+Then the supergraph schema would require both scopes for it.
```

Comment:
Interesting- @chandrikas this is related to our earlier conversation in Slack.

Reply:
Yes, for now I think it is an AND (we've communicated this in our RFC as well). We need to make sure we keep a door open for allowing ORing of directives.